peer/brontide_test.go

```diff
@@ -1340,3 +1340,217 @@ func TestHandleRemovePendingChannel(t *testing.T) {
 		})
 	}
 }
+
+func TestPeerPongCorrect(t *testing.T) {
+	t.Parallel()
+
+	mockConn := newMockConn(t, 10)
+
+	writeBufferPool := pool.NewWriteBuffer(
+		pool.DefaultWriteBufferGCInterval,
+		pool.DefaultWriteBufferExpiryInterval,
+	)
+	writePool := pool.NewWrite(
+		writeBufferPool, 1, timeout,
+	)
+	require.NoError(t, writePool.Start())
+
+	readBufferPool := pool.NewReadBuffer(
+		pool.DefaultReadBufferGCInterval,
+		pool.DefaultReadBufferExpiryInterval,
+	)
+
+	readPool := pool.NewRead(
+		readBufferPool, 1, timeout,
+	)
+	require.NoError(t, readPool.Start())
+
+	rawFeatures := lnwire.NewRawFeatureVector()
+	rawFeatures.Set(lnwire.DataLossProtectRequired)
+	features := lnwire.NewFeatureVector(rawFeatures, lnwire.Features)
+	p := NewBrontide(Config{
+		Conn:           mockConn,
+		Features:       features,
+		LegacyFeatures: features,
+		WritePool:      writePool,
+		ReadPool:       readPool,
+	})
+
+	err := p.handleInitMsg(lnwire.NewInitMessage(rawFeatures, rawFeatures))
+	require.NoError(t, err)
+
+	newPingPayload := func() []byte {
+		return []byte{0, 0, 0, 0}
+	}
+	p.wg.Add(4)
+	go p.queueHandler()
+	go p.writeHandler()
+	go p.readHandler()
+	go p.pingHandler(
+		newPingPayload,
+		func() uint16 { return 4 },
+		2*time.Second,
+		time.Second,
+	)
+
+	// we have to signal that the brontide is ready before the disconnect
+	// will proceed
+	close(p.startReady)
+
+	pong := make([]byte, 0, 256)
+	buf := bytes.NewBuffer(pong)
+	<-mockConn.writtenMessages
+
+	_, err = lnwire.WriteMessage(
+		buf,
+		lnwire.Message(&lnwire.Pong{PongBytes: []byte{0, 0, 0, 0}}),
+		0,
+	)
+	require.NoError(t, err)
+
+	mockConn.readMessages <- buf.Bytes()
+	time.Sleep(time.Second)
+	require.False(t, mockConn.closed)
+
+	close(p.quit)
```

Comment:
this causes a panic though?

Reply:
oops yeah that is handled by the disconnect